### PR TITLE
CFE-1168: Add CMPSMachineNamePrefix feature-gate

### PR DESCRIPTION
**Pull request by chiragkyal**
Add new `CMPSMachineNamePrefix` feature-gate to guard the implementation changes done to support customized control plane machine names.

- Part of [CFE-1168](https://issues.redhat.com/browse/CFE-1168)
- Implements https://github.com/openshift/enhancements/pull/1714

### CODE REVIEW

File: features/features.go [hunk 1]
```diff
@@ -219,6 +219,14 @@ var (
 						enableIn(configv1.DevPreviewNoUpgrade, configv1.TechPreviewNoUpgrade).
 						mustRegister()
 
+	FeatureGateCMPSMachineNamePrefix = newFeatureGate("CMPSMachineNamePrefix").
+						reportProblemsToJiraComponent("cloud-provider").
```
Comment (JoelSpeed) [inline features/features.go:223]:
```suggestion
						reportProblemsToJiraComponent("Cloud Compute / ControlPlaneMachineSet").
```

**Review by JoelSpeed — COMMENTED** [review]
Need to make sure to use the correct jira bug component for this FG

File: features/features.go (hunk 1 above)
Comment (chiragkyal) [reply features/features.go]: Thanks for the suggestion. I've updated the Jira component.

**Comment by chiragkyal** [comment]
@JoelSpeed this can also be merged, I believe.

File: features/features.go [hunk 2]
```diff
@@ -219,6 +219,14 @@ var (
 						enableIn(configv1.DevPreviewNoUpgrade, configv1.TechPreviewNoUpgrade).
 						mustRegister()
 
+	FeatureGateCMPSMachineNamePrefix = newFeatureGate("CMPSMachineNamePrefix").
```
Comment (damdo) [inline features/features.go:222]: There is a typo here, it would be CPMS rather than CMPS. Same issue in the other places around here.

Comment (chiragkyal) [reply features/features.go:222]: My bad! Thanks for the catch. I'll get them fixed in https://github.com/openshift/api/pull/2086
Copy/paste wasn't a good idea :(

**Pull request merged**